jmxtrans-output/jmxtrans-output-gelf/src/main/java/com/googlecode/jmxtrans/model/output/gelf/GelfWriter.java

```diff
@@ -0,0 +1,220 @@
+			final GelfTransports gelfTransports;
+
+			switch (transport.toUpperCase()) {
+				case "TCP": gelfTransports = GelfTransports.TCP;
+					break;
+				case "UDP": gelfTransports = GelfTransports.UDP;
+					break;
+				case "TEST": useMock = true;
+				default: gelfTransports = GelfTransports.TCP;
+			}
+
+			gelfConfiguration.transport(gelfTransports);
+		}
+
+		if (queueSize != null) {
+			gelfConfiguration.queueSize(queueSize);
+		}
+
+		if (connectTimeout != null) {
+			gelfConfiguration.connectTimeout(connectTimeout);
+		}
+
+		if (reconnectDelay != null) {
+			gelfConfiguration.reconnectDelay(reconnectDelay);
+		}
+
+		if (tcpNoDelay != null) {
+			gelfConfiguration.tcpNoDelay(tcpNoDelay);
+		}
+
+		if (sendBufferSize != null) {
+			gelfConfiguration.sendBufferSize(sendBufferSize);
+		}
+
+		if (tlsEnabled != null) {
+			if (tlsEnabled) {
+				gelfConfiguration.enableTls();
+			} else {
+				gelfConfiguration.disableTls();
+			}
+		}
+
+		if (tlsCertVerificationEnabled != null) {
+			if (tlsCertVerificationEnabled) {
+				gelfConfiguration.enableTlsCertVerification();
+			} else {
+				gelfConfiguration.disableTlsCertVerification();
+			}
+		}
+
+		if (tlsTrustCertChainFile != null) {
+			gelfConfiguration.tlsTrustCertChainFile(new File(tlsTrustCertChainFile));
+		}
+
+		if (maxInflightSends != null) {
+			gelfConfiguration.maxInflightSends(maxInflightSends);
+		}
+
+		log.debug("Created gelf configuration: %s", gelfConfiguration.toString());
+
+	}
+
+	@Override
+	public void validateSetup(final Server server, final Query query)
+		throws ValidationException
+	{
+		// no validations
+	}
+
+	@Override
+	protected void internalWrite(
+		final Server server,
+		final Query query,
+		final ImmutableList<Result> results) throws Exception
+	{
+		final List<String> typeNames = this.getTypeNames();
+
+		final GelfTransport transport;
+
+		if (useMock) {
+			transport = new MockGelfTransport();
+		} else {
+			transport = GelfTransports.create(this.gelfConfiguration);
+		}
+
+		final GelfMessageBuilder messageBuilder = new GelfMessageBuilder(
+			"",
+			server.getHost()
+		);
+
+		final List<String> messages = new ArrayList<>();
+
+		for (final Result result : results) {
+			log.debug("Query result: [{}]", result);
+			for (final Map.Entry<String, Object> values: result.getValues().entrySet()) {
+				final String key = KeyUtils.getKeyString(query, result, values, typeNames);
+
+				messages.add(
+					String.format(
+						"%s=%s",
+						key,
+						values.getValue()
+					)
+				);
+
+				messageBuilder.additionalField(key, values.getValue());
+			}
+		}
+
+		if (additionalFields != null) {
+			for (final Map.Entry<String, Object> additionalField : additionalFields
+				.entrySet()) {
+				messageBuilder.additionalField(additionalField.getKey(),
+					additionalField.getValue());
+			}
+		}
+
+		final String message = StringUtils.join(messages, " ");
+
+		messageBuilder.message(message);
+		messageBuilder.fullMessage(message);
+
+		log.debug(
```

Comment:
Don't use `String.format()` in log messages, use the [SLF4J placeholders](https://www.slf4j.org/manual.html#typical_usage) (`{}`).